src/main/scala/it/pagopa/interop/authorizationmanagement/model/persistence/KeyAdapters.scala

```diff
@@ -1,22 +1,29 @@
 package it.pagopa.interop.authorizationmanagement.model.persistence
 
-import it.pagopa.interop.authorizationmanagement.errors.KeyManagementErrors.ThumbprintCalculationError
+import it.pagopa.interop.authorizationmanagement.errors.KeyManagementErrors.{InvalidKey, ThumbprintCalculationError}
 import it.pagopa.interop.authorizationmanagement.model.key.{Enc, PersistentKey, PersistentKeyUse, Sig}
 import it.pagopa.interop.authorizationmanagement.jwk.model.Models._
 import it.pagopa.interop.authorizationmanagement.model.{KeyUse, Key, OtherPrimeInfo, JWKKey}
 import it.pagopa.interop.authorizationmanagement.jwk.converter.KeyConverter
 import it.pagopa.interop.authorizationmanagement.model.KeyUse.{ENC, SIG}
-import java.util.UUID
+import cats.syntax.all._
 
 object KeyAdapters {
 
   implicit class PersistentKeyWrapper(private val p: PersistentKey) extends AnyVal {
-    def toApi: Either[Throwable, Key] =
-      KeyConverter
-        .fromBase64encodedPEMToAPIKey(p.kid, p.encodedPem, p.use.toJwk, p.algorithm)
-        .map(_ =>
-          Key(p.userId.getOrElse(UUID.randomUUID()), p.kid, p.name, p.encodedPem, p.algorithm, p.use.toApi, p.createdAt)
-        )
+    def toApi: Either[Throwable, Key] = {
+      val apiKey = KeyConverter.fromBase64encodedPEMToAPIKey(p.kid, p.encodedPem, p.use.toJwk, p.algorithm)
+      p.userId match {
+        case Some(userId) =>
+          apiKey.map(_ => Key(userId, p.kid, p.name, p.encodedPem, p.algorithm, p.use.toApi, p.createdAt))
+        case None         =>
+          p.relationshipId match {
+            case Some(relationshipId) =>
+              apiKey.map(_ => Key(relationshipId, p.kid, p.name, p.encodedPem, p.algorithm, p.use.toApi, p.createdAt))
+            case None => InvalidKey(p.kid, s"User Id and Relationship Id not found on key ${p.kid}").asLeft[Key]
+          }
+      }
+    }
```

Comment:
correct me if I'm wrong, but after the migration we expect the `userId` to be always present.
Replacing it with `relationshipId` would cause confusion for the client who receive the response.

We could simplify the function as
```suggestion
    def toApi: Either[Throwable, Key] = for {
        userId <- p.userId.toEither(MissingUserId) // This would cause a 500
        _ <- KeyConverter.fromBase64encodedPEMToAPIKey(p.kid, p.encodedPem, p.use.toJwk, p.algorithm)
      } yield Key(userId, p.kid, p.name, p.encodedPem, p.algorithm, p.use.toApi, p.createdAt)
    }
```

Reply:
Yes I agree.
But UserId and RelationShipId are both Option, we can find both in the None state. Therefore you say that almost one must be present, and after the migration if UserId is not present we need to return an HTTP 500?

Reply:
correct, this is my understanding.

@beetlecrunch please confirm, do we expect edge cases when both are missing?

Reply:
If exists a `relationship` exists a `userId` too.
So excluding some bugs in the migration script, `userId` should always be set.